Add json file for Xarray Badge

ianhi opened the pull request:
Add a json file for badge to the main repo (a la ruff)
See https://github.com/pydata/xarray/issues/10593

Here is it rendered: [![image](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/ianhi/xarray/refs/heads/badge/doc/badge.json)](xarray.dev)
- [x] Closes https://github.com/pydata/xarray/issues/10593
- [NA] Tests added
- [NA] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [NA] New functions/methods are listed in `api.rst`

ianhi commented:
I'm worried that the white text on that background does not pass the accessibilty contrast guidelines but im a little confused because 4a4a4a seems to be a much grayer color so im not sure why the text is showing up white

VeckoTheGecko commented:
> 4a4a4a seems to be a much grayer color

4a4a4a is for the left side (labelColor) not the font
> labelColor string
> Background color of the left part (hex, rgb, rgba, hsl, hsla and css named colors supported).
> 
> Example: abcdef

Font colour isn't configurable https://github.com/badges/shields/issues/504#issuecomment-717563006 - best to just choose a different background colour if it doesn't meet accessibility requirements I think

ianhi commented:
I moved it to the doc directory, and also updated the colors to be accessible: [![image](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/ianhi/xarray/refs/heads/badge/doc/badge.json)](xarray.dev)
